Add files via upload

excitedsoup opened the pull request.

excitedsoup committed:
Add files via upload

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+def main():
+    f = open('WarAndPeace.txt', 'r', encoding="utf8")
```
would prefer that this file not be included in the PR and instead be downloaded by the script

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    
+    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
+    
+    for i in text:
```
nice understanding of strings in python.  but it's bad practice to use `i` for non-counting integers.  would prefer something more descriptive like `symbol`

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    
+    splitText = text.split()
+    
+    for i in range(len(splitText)):
```
this can be simplified as `for word in splitText`

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    splitText = text.split()
+    
+    for i in range(len(splitText)):
+        splitText[i] = splitText[i].lower()
```
and this can be simplified as `splitText = [w.lower() for w in splitText]`

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    
+    for i in text:
+        if i in punct:
+            text = text.replace(i, ' ')
```
this is an **incredibly** slow loop, and probably doesn't do exactly what you want it to be doing

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    uniqueWords = set(splitText)
+    
+    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
+main()
```
this should be wrapped in an `if __name__ == '__main__':`

enkeboll commented on hw01-tolstoy.py:
```diff
@@ -0,0 +1,20 @@
+    
+    uniqueWords = set(splitText)
+    
+    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
```
your solution gets 19k words, which is a little low I think

enkeboll reviewed (approved):
Good implementation, but losing a point for following directions-- should be in directory `homework/`
9/10

excitedsoup commented:
Yeah, you're right - I partially pulled the implementation from a previous
project, since I had to strip punctuation before (which had a whole
headache involving a different variation of double quote characters not
included in string.punctuation), and I'm not entirely sure why I made that
loop that way. Porting it to what I'd used in that previous project looks
like:

text = text.translate(str.maketrans('','', punct))

which makes its runtime less than a second and gives a result of 21015
words. I didn't exactly follow what the " if __name__ == '__main__': " was
talking about - where should that be and what does that do?

I appreciate the in-depth feedback.

Thanks,
Peter

On Tue, Sep 14, 2021 at 1:59 AM Andy Enkeboll ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Good implementation, but losing a point for following directions-- should
> be in directory homework/
> 9/10
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707931263>:
>
> > @@ -0,0 +1,20 @@
>
> +def main():
>
> +    f = open('WarAndPeace.txt', 'r', encoding="utf8")
>
>
> would prefer that this file not be included in the PR and instead be
> downloaded by the script
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707931582>:
>
> > @@ -0,0 +1,20 @@
>
> +def main():
>
> +    f = open('WarAndPeace.txt', 'r', encoding="utf8")
>
> +    text = f.read()
>
> +    f.close()
>
> +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
>
> nice understanding of strings in python. but it's bad practice to use i
> for non-counting integers. would prefer something more descriptive like
> symbol
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707931809>:
>
> > @@ -0,0 +1,20 @@
>
> +def main():
>
> +    f = open('WarAndPeace.txt', 'r', encoding="utf8")
>
> +    text = f.read()
>
> +    f.close()
>
> +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
> +        if i in punct:
>
> +            text = text.replace(i, ' ')
>
> +
>
> +    splitText = text.split()
>
> +
>
> +    for i in range(len(splitText)):
>
>
> this can be simplified as for word in splitText
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707932040>:
>
> > @@ -0,0 +1,20 @@
>
> +def main():
>
> +    f = open('WarAndPeace.txt', 'r', encoding="utf8")
>
> +    text = f.read()
>
> +    f.close()
>
> +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
> +        if i in punct:
>
> +            text = text.replace(i, ' ')
>
> +
>
> +    splitText = text.split()
>
> +
>
> +    for i in range(len(splitText)):
>
> +        splitText[i] = splitText[i].lower()
>
>
> and this can be simplified as splitText = [w.lower() for w in splitText]
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707934269>:
>
> > @@ -0,0 +1,20 @@
>
> +def main():
>
> +    f = open('WarAndPeace.txt', 'r', encoding="utf8")
>
> +    text = f.read()
>
> +    f.close()
>
> +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
> +        if i in punct:
>
> +            text = text.replace(i, ' ')
>
>
> this is an *incredibly* slow loop, and probably doesn't do exactly what
> you want it to be doing
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707934490>:
>
> > +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
> +        if i in punct:
>
> +            text = text.replace(i, ' ')
>
> +
>
> +    splitText = text.split()
>
> +
>
> +    for i in range(len(splitText)):
>
> +        splitText[i] = splitText[i].lower()
>
> +
>
> +    uniqueWords = set(splitText)
>
> +
>
> +    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
>
> +main()
>
>
> this should be wrapped in an if __name__ == '__main__':
> ------------------------------
>
> In hw01-tolstoy.py
> <https://github.com/excitedsoup/DATA690/pull/2#discussion_r707934575>:
>
> > +    f.close()
>
> +
>
> +    punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~—“”'
>
> +
>
> +    for i in text:
>
> +        if i in punct:
>
> +            text = text.replace(i, ' ')
>
> +
>
> +    splitText = text.split()
>
> +
>
> +    for i in range(len(splitText)):
>
> +        splitText[i] = splitText[i].lower()
>
> +
>
> +    uniqueWords = set(splitText)
>
> +
>
> +    print('There are ' + str(len(uniqueWords)) + ' unique words in the provided text file of War and Peace')
>
>
> your solution gets 19k words, which is a little low I think
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/excitedsoup/DATA690/pull/2#pullrequestreview-753497693>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APV5DAH5BBQAUIZEUWO4MXTUB3QEJANCNFSM5DTFTXMA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

enkeboll commented:
RE: `if __name__ == '__main__'`, read this answer which explains it better than I can:
https://stackoverflow.com/questions/419163/what-does-if-name-main-do

tl;dr: it's a good idea to do like you did, and encapsulate your code in a function first.  It's then **great** to wrap that function call in an if statement that ensures you only call that function exactly when you intend to.  It's just good a practice to get in the habit of doing!
